src/common/client.js

```diff
@@ -0,0 +1,16 @@
+import ky from 'ky';
+
+const credentials = 'include';
+
+const headers = {
+  'content-type': 'application/json; charset=utf-8',
+  accept: 'application/json',
+};
+
+function client(base, options = {}) {
+  return ky.extend({
+    credentials, headers, prefixUrl: base, ...options,
```

Comment:
On the way to the office, I had a more general thought… (Sorry for bringing up so much here, it's just that this is the first adapter we're moving over to use an API and to work asynchronously so naturally there's a lot of new things to think about 🤗).

Previously, all our adapters were synchronous, so the adapters and thus the UI would return results very quickly.

When we're doing async requests, these may take a moment longer which means Tickety-Tick will not be showing anything instantly. This means to things:

1. We should update the UI to display an **activity indicator** while adapters are scanning for tickets (could be a separate PR)
2. We should potentially **enforce timeouts** on adapters (can also be dealt with in a separate issue, but let's think it through)

Currently in `src/common/search.js` we simply use `Promise.all` to wait for all adapters to finish, meaning the slowest adapter determines how long we wait. Usually we only expect only one adapter to return results though. Therefore [`Promise.race`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/race) might actually be a better fit.

The simplest but not necessarily most robust approach might be to just set a default timeout for the `ky` client instances, e.g. 1000 ms?

This also leads to the question of how to deal with failed requests and rejected promises. In the simplest case, we don't handle this at all and trust all adapters to never fail and to resolve with `null` instead so they don't disturb the search. Or we could wrap all adapter promises in `search.js` and make sure they simply resolve with `null` in case they're rejected/throw an error.

Reply:
Ah… `Promise.race` doesn't quite fit by the way, because what we need is the first non-rejected, non-null result whereas `Promise.race` just resolves or rejects with the result of the promise that finishes first, no matter whether it was resolved or rejected. We can find a solution anyway 🙂

Reply:
@pmeinhardt I set the timeout for the HTTP client, I guess we can address the other points in a separated PR.